crypto/data_formats.go

```diff
@@ -53,3 +55,66 @@ func HashLogRoot(root trillian.SignedLogRoot) ([]byte, error) {
 	}
 	return hash[:], nil
 }
+
+// LogRootV1 contains the fields verified by SignedLogRoot
```

Comment:
Having these type definitions in `crypto/data_formats.go` feels a bit out of the way, given that this is a data type that forms part of the external API to Trillian.  It's not a proto structure so we can't directly have it in the top-level .proto files that define the API, but we could do something like:
 - Add a top-level `types.go` file with these API types as Go types.
 - Add (in a later PR) comments to the .proto files that say something like:
   ```
   // The log_root field holds the TLS-serialization of the following 
   // structure (described in RFC5246 notation):
   //  struct {
   //    uint16 version;
   //    uint64 tree_size;
   //    opaque root_hash<0..128>;
   //    uint64 timestamp_nanos;
   //    uint64 revision;
   //    opaque metadata<0..65535>;
   //  } LogRootV1;
   ```

That way the .proto files would contain an unambiguous description of the API details, and a non-Go user of Trillian would have enough information to be able to use the API.

Reply:
I agree on both points. Unfortunately, I can't add non-proto code to a proto package. 
I'm tempted to move the proto definitions to `api/v1/trillian_proto` so we can use the top level `trillian` package name for this type of stuff.

Reply:
In the following pull requests, I'll be defining methods to verify `SignedLogRoot` that will return `LogRootV1` upon verification.  That was the rational for keeping these methods in the `crypto` package. Should I create a `crypto/signature` package?